actionpack/lib/action_dispatch/journey/router.rb

```diff
@@ -47,16 +47,16 @@ def serve(req)
 
           req.path_parameters = tmp_params
 
-          _, headers, _ = response = route.app.serve(req)
+          response = Rack::Response[*route.app.serve(req)]
 
-          if "pass" == headers["X-Cascade"]
+          if "pass" == response.headers["X-Cascade"]
```

Comment:
For compatibility, this makes sense, but assuming we go all in with Rack 3 on Rails 8, this will no longer be needed (since lower case normalisation is a requirement of Rack 3). I would suggest adding a "TODO" in relation to this so we know to follow up.